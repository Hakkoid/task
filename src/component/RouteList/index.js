import React, { Component , Fragment} from 'react';
import {SortableContainer, SortableElement} from 'react-sortable-hoc';
import Point from '../Point';
import Add from '../Add';
import List from '../List';
import { connect } from "react-redux";
import { addPoint, removePoint, swapPoint } from '../../redux/actions';



const SortablePoint = SortableElement(Point);

const SortableList = SortableContainer(List);

class RouteList extends Component {
  constructor(props){
    super(props);
    this.handleSortEnd = this.handleSortEnd.bind(this);
    this.handleRemove = this.handleRemove.bind(this);
    this.handleAddPoint = this.handleAddPoint.bind(this);
  }

  handleSortEnd({oldIndex, newIndex}){
    this.props.swapPoint({oldIndex, newIndex})
  }

  handleRemove(id){
    this.props.removePoint(id);
  }

  handleAddPoint(text){
    if( text === ""){
      return
    }

    this.props.addPoint({ text, coordinates: this.props.center});
  }

  render() {

    const points = this.props.points.map((point, index)=>{
      return (
        <SortablePoint id={point.id} index={index} key={point.id} onClose={this.handleRemove}>
          {point.text}
        </SortablePoint>
      )
    })

    return (
      <Fragment>
        <Add onAdd = {this.handleAddPoint} />
        <SortableList distance={1} helperClass="Point_focus" onSortEnd={this.handleSortEnd}>
          {points}
        </SortableList>
      </Fragment>
    );
  }
}


const mapStateToProps = (state, ownProps) => {
  const { points, center } = state;
  return { points, center }
}

export default connect(
  mapStateToProps,
  { addPoint, removePoint, swapPoint }
)(RouteList)