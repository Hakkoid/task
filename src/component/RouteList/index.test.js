import React from 'react';
import { mount } from 'enzyme';
import RouteList from './index';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../../store.js';


describe( 'RouteList', () => {

    it('renders without crashing', () => {
        mount(
            <Provider store = { store }>
                <RouteList />
            </Provider>
        )
    })

    it( 'Snapshot', () => {
        const component = renderer.create(
            <Provider store = { store }>
                <RouteList />
            </Provider>,
        );

        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    })
})