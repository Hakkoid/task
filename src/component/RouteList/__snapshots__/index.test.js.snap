// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RouteList Snapshot 1`] = `
Array [
  <form
    className="Add"
    onSubmit={[Function]}
  >
    <input
      className="Add__Input"
      onChange={[Function]}
      type="text"
      value=""
    />
    <input
      className="Add__Button"
      type="submit"
      value="Add"
    />
  </form>,
  <ul
    className="List"
  />,
]
`;
